e2e/pages/manageCases/caseProgression/completeTheOrder/adminEditAndApproveAnOrder4Page.ts

```diff
@@ -0,0 +1,71 @@
+import { Page } from "@playwright/test";
+import AccessibilityTestHelper from "../../../../common/accessibilityTestHelper";
+import { Selectors } from "../../../../common/selectors";
+import { AdminEditAndApproveAnOrder4Content } from "../../../../fixtures/manageCases/caseProgression/completeTheOrder/adminEditAndApproveAnOrder4Content";
+import { Helpers } from "../../../../common/helpers";
+import { CommonStaticText } from "../../../../common/commonStaticText";
+
+enum UniqueSelectors {
+  editOrderNo = "#doYouWantToEditTheOrder_No",
+}
+export class AdminEditAndApproveAnOrder4Page {
+  public static async adminEditAndApproveAnOrder4Page(
+    page: Page,
+    accessibilityTest: boolean,
+  ): Promise<void> {
+    await this.checkPageLoads(page, accessibilityTest);
+    await this.fillInFields(page);
+    await this.continue(page);
+  }
+
+  private static async checkPageLoads(
+    page: Page,
+    accessibilityTest: boolean,
+  ): Promise<void> {
+    await page
+      .locator(`${Selectors.GovukHeadingL}`, {
+        hasText: `${AdminEditAndApproveAnOrder4Content.govUkHeadingL}`,
+      })
+      .waitFor();
+    await Promise.all([
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${Selectors.GovukFormLabel}:text-is("${AdminEditAndApproveAnOrder4Content.formLabel1}")`,
+        1,
+      ),
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${Selectors.headingH3}:text-is("${AdminEditAndApproveAnOrder4Content.headingH3}")`,
+        1,
+      ),
+      Helpers.checkVisibleAndPresent(
```

Comment:
You may have to do a switch case here to look for all the potential options and use the "createOrderFL401Options" type for it